server/client_manager.py

```diff
@@ -276,55 +240,60 @@ def disemvowel_message(self, message):
 
     def __init__(self, server):
         self.clients = set()
-        self.cur_id = 0
         self.server = server
+        self.cur_id = [False] * self.server.config['playerlimit']
+        self.clients_list = []
 
     def new_client(self, transport):
-        c = self.Client(self.server, transport, self.cur_id)
+        cur_id = 0
+        for i in range(self.server.config['playerlimit']):
+                if not self.cur_id[i]:
+                    cur_id = i
+                    break
+        c = self.Client(self.server, transport, cur_id, self.server.get_ipid(transport.get_extra_info('peername')[0]))
         self.clients.add(c)
-        self.cur_id += 1
+        self.cur_id[cur_id] = True
```

Comment:
ok, now I'm understanding what you want to do.. you basically want to take the lowest unoccupied ID right?